TP ESPECIAL FUNDAMENTOS DE LA CIENCIA DE DATOS 2024 

DATASET: CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2022

INTEGRANTES:
Abril Valentina Valentina Juarez, Matias Müller Gonzales y Julián Elias Rivero

In [2]:
import pandas as pd

In [3]:
pd.__version__ #es necesario?

'2.2.2'

In [4]:
raw_dataset = pd.read_csv("Calidad_de_agua_2022.csv", sep =";")

Desplegamos el dataset para realizar un vistazo generico

In [5]:
raw_dataset

orden                                 sitios codigo       fecha   año  \
0        1           Canal Villanueva y Río Luján  TI001   23/2/2022  2022   
1        2           Río Lujan y Arroyo Caraguatá  TI006   23/2/2022  2022   
2        3            Canal Aliviador y Río Lujan  TI002   23/2/2022  2022   
3        4  Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022  2022   
4        5            Río Reconquista y Río Lujan  TI004   23/2/2022  2022   
..     ...                                    ...    ...         ...   ...   
163     38                  Balneario Palo Blanco  BS092  31/10/2022  2022   
164     39          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  2022   
165     40                     Playa La Bagliardi  BS091  31/10/2022  2022   
166     41                    Balneario Municipal  BS094  31/10/2022  2022   
167     42                      Playa La Balandra  BS093  31/10/2022  2022   

       campaña tem_agua tem_aire           od           ph  ...     dbo_mg_l  \
0       Verano     24.5     23.3          5.3         6.56  ...          6.2   
1       Verano     25.4     23.3         2.25         6.56  ...          5.8   
2       Verano     24.6     23.3         2.94         6.59  ...          1.9   
3       Verano     25.2     23.3         2.22         7.45  ...          5.8   
4       Verano     24.1       20         1.02         6.39  ...          2.6   
..         ...      ...      ...          ...          ...  ...          ...   
163  Primavera       10       12  no se midió  no se midió  ...  no se midió   
164  Primavera       10       12  no se midió  no se midió  ...  no se midió   
165  Primavera       10       12  no se midió  no se midió  ...  no se midió   
166  Primavera       10       12  no se midió  no se midió  ...  no se midió   
167  Primavera       10       12  no se midió  no se midió  ...  no se midió   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                <0.10        <0.005        <0.001   
1         29            34                <0.10        <0.005        <0.001   
2         29            17                <0.10        <0.005        <0.001   
3         29            23                <0.10        <0.005        <0.001   
4         59            18                <0.10        <0.005        <0.001   
..       ...           ...                  ...           ...           ...   
163      <30            23                <0.10        <0.005        <0.001   
164       31          18.2                <0.10        <0.005        <0.001   
165      <30            40                <0.10        <0.005        <0.001   
166       39            90                <0.10             5        <0.001   
167      <30            70                <0.10             5        <0.001   

    clorofila_a_ug_l microcistina_ug_l ica             calidad_de_agua  
0                <10             <0.20  55             Muy deteriorada  
1                <10             <0.20  42  Extremadamente deteriorada  
2                <10               0.2  45             Muy deteriorada  
3                <10             <0.20  46             Muy deteriorada  
4                <10             <0.20  44  Extremadamente deteriorada  
..               ...               ...  ..                         ...  
163              2.1             <0.15  43  Extremadamente deteriorada  
164             20.2             <0.15  37  Extremadamente deteriorada  
165              0.2             <0.15  49             Muy deteriorada  
166             10.5             <0.15  39  Extremadamente deteriorada  
167             48.0             <0.15  34  Extremadamente deteriorada  

[168 rows x 31 columns]

Ahora le pedimos un poco de informacion para analizar contexto de los datos

In [6]:
raw_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168 entries, 0 to 167
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   orden                    168 non-null    int64 
 1   sitios                   168 non-null    object
 2   codigo                   168 non-null    object
 3   fecha                    168 non-null    object
 4   año                      168 non-null    object
 5   campaña                  168 non-null    object
 6   tem_agua                 168 non-null    object
 7   tem_aire                 166 non-null    object
 8   od                       168 non-null    object
 9   ph                       168 non-null    object
 10  olores                   168 non-null    object
 11  color                    168 non-null    object
 12  espumas                  168 non-null    object
 13  mat_susp                 168 non-null    object
 14  colif_fecales_ufc_100ml  167 non-null    o

En total tenemos 31 columnas (atributos) y 168 filas (observaciones). Se puede ver que hay ciertas columnas con NULL y que el tipo de datos practicamente es OBJECT exceptuando la primer columna "orden" que es de tipo INT64.

In [7]:
raw_dataset.head()

orden                                 sitios codigo      fecha   año  \
0      1           Canal Villanueva y Río Luján  TI001  23/2/2022  2022   
1      2           Río Lujan y Arroyo Caraguatá  TI006  23/2/2022  2022   
2      3            Canal Aliviador y Río Lujan  TI002  23/2/2022  2022   
3      4  Río Carapachay y Arroyo Gallo Fiambre  TI003  23/2/2022  2022   
4      5            Río Reconquista y Río Lujan  TI004  23/2/2022  2022   

  campaña tem_agua tem_aire    od    ph  ... dbo_mg_l dqo_mg_l turbiedad_ntu  \
0  Verano     24.5     23.3   5.3  6.56  ...      6.2       29            90   
1  Verano     25.4     23.3  2.25  6.56  ...      5.8       29            34   
2  Verano     24.6     23.3  2.94  6.59  ...      1.9       29            17   
3  Verano     25.2     23.3  2.22  7.45  ...      5.8       29            23   
4  Verano     24.1       20  1.02  6.39  ...      2.6       59            18   

  hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l clorofila_a_ug_l  \
0                <0.10        <0.005        <0.001              <10   
1                <0.10        <0.005        <0.001              <10   
2                <0.10        <0.005        <0.001              <10   
3                <0.10        <0.005        <0.001              <10   
4                <0.10        <0.005        <0.001              <10   

  microcistina_ug_l ica             calidad_de_agua  
0             <0.20  55             Muy deteriorada  
1             <0.20  42  Extremadamente deteriorada  
2               0.2  45             Muy deteriorada  
3             <0.20  46             Muy deteriorada  
4             <0.20  44  Extremadamente deteriorada  

[5 rows x 31 columns]

A simple vista veo que tendremos que tratar lo siguiente:
1) Hay valores que enves de tener un tipo de dato cuantitativo en el atributo, tienen un string categorico que dice "no se midio", por ejemplo en OD(oxigeno disuelto), esto deberia de ser reemplazado (ver despues con que).
2) Hay datos cuantitativos que adelante del valor numerico tienen un string "<", como en el caso de las mediciones de los contaminantes en el agua.
Podriamos de alguna manera, mediante una funcion operar y captar ese string para borrar todas las ocurrencias. 
3) Hay datos cualitativos nominales que indican "presencia" o "ausencia" de algun compuesto en el agua, pueden ser reemplazados por {0, 1} de acuerdo a su impacto en la muestra.
4) Hay datos cualitativos ordinales como "muy deteriorada" o "extremadamente deteriorada" que hacen referencia al agua de esa zona, podrian ser reemplazados mediante transformacion "ordinal encoding" ya que el orden es significativo. Algo del estilo... Rating: {low, medium, high} → {1,2,3}

Veamos cuantos valores repetidos tiene el dataset

In [8]:
raw_dataset.duplicated().sum()

np.int64(0)

Por lo visto no existe valores repetidos a simple vista (continuar con la exploracion de valores duplicados)

Ya que empezaremos a modificar el dataset, una buena practica es copiar el dataset

In [9]:
copy_dataset = raw_dataset.copy()

podemos eliminar columnas que no tienen mucho sentido, como por ejemplo "orden" y vamos a eliminar "año", ya que todas(revisar que hay 5 observaciones que tienen "no se midio") las observaciones se hicieron en 2022 y no agregan mucha informacion

In [10]:
copy_dataset = copy_dataset.drop("orden",axis=1)
copy_dataset = copy_dataset.drop("año",axis=1)
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire           od           ph     olores      color  ...  \
0       24.5     23.3          5.3         6.56   Ausencia   Ausencia  ...   
1       25.4     23.3         2.25         6.56  Presencia  Presencia  ...   
2       24.6     23.3         2.94         6.59   Ausencia  Presencia  ...   
3       25.2     23.3         2.22         7.45  Presencia  Presencia  ...   
4       24.1       20         1.02         6.39   Ausencia  Presencia  ...   
..       ...      ...          ...          ...        ...        ...  ...   
163       10       12  no se midió  no se midió   Ausencia   Ausencia  ...   
164       10       12  no se midió  no se midió   Ausencia  Presencia  ...   
165       10       12  no se midió  no se midió   Ausencia   Ausencia  ...   
166       10       12  no se midió  no se midió   Ausencia   Ausencia  ...   
167       10       12  no se midió  no se midió   Ausencia   Ausencia  ...   

        dbo_mg_l dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l  \
0            6.2       29            90                <0.10        <0.005   
1            5.8       29            34                <0.10        <0.005   
2            1.9       29            17                <0.10        <0.005   
3            5.8       29            23                <0.10        <0.005   
4            2.6       59            18                <0.10        <0.005   
..           ...      ...           ...                  ...           ...   
163  no se midió      <30            23                <0.10        <0.005   
164  no se midió       31          18.2                <0.10        <0.005   
165  no se midió      <30            40                <0.10        <0.005   
166  no se midió       39            90                <0.10             5   
167  no se midió      <30            70                <0.10             5   

    cd_total_mg_l clorofila_a_ug_l microcistina_ug_l ica  \
0          <0.001              <10             <0.20  55   
1          <0.001              <10             <0.20  42   
2          <0.001              <10               0.2  45   
3          <0.001              <10             <0.20  46   
4          <0.001              <10             <0.20  44   
..            ...              ...               ...  ..   
163        <0.001              2.1             <0.15  43   
164        <0.001             20.2             <0.15  37   
165        <0.001              0.2             <0.15  49   
166        <0.001             10.5             <0.15  39   
167        <0.001             48.0             <0.15  34   

                calidad_de_agua  
0               Muy deteriorada  
1    Extremadamente deteriorada  
2               Muy deteriorada  
3               Muy deteriorada  
4    Extremadamente deteriorada  
..                          ...  
163  Extremadamente deteriorada  
164  Extremadamente deteriorada  
165             Muy deteriorada  
166  Extremadamente deteriorada  
167  Extremadamente deteriorada  

[168 rows x 29 columns]

Ahora vamos a ver que columnas tienen nulos

In [11]:
copy_dataset.isna().sum()

sitios                      0
codigo                      0
fecha                       0
campaña                     0
tem_agua                    0
tem_aire                    2
od                          0
ph                          0
olores                      0
color                       0
espumas                     0
mat_susp                    0
colif_fecales_ufc_100ml     1
escher_coli_ufc_100ml       1
enteroc_ufc_100ml           1
nitrato_mg_l                0
nh4_mg_l                    0
p_total_l_mg_l              0
fosf_ortofos_mg_l           0
dbo_mg_l                    1
dqo_mg_l                    0
turbiedad_ntu               0
hidr_deriv_petr_ug_l        0
cr_total_mg_l               0
cd_total_mg_l               0
clorofila_a_ug_l            1
microcistina_ug_l           0
ica                        13
calidad_de_agua            14
dtype: int64

Pareciera que existen muy pocos nulos, pero vamos a ver que sucede con el "no se midió" , asi que vamos a mostrar algunos ejemplos

In [12]:
copy_dataset[copy_dataset["dbo_mg_l"] == "no se midió"] #poner algun otro ejemplo

sitios    codigo        fecha  \
10                       Espigón La Farola     SI021    23/2/2022   
12                Playa Espigón de Pacheco     SI024    23/2/2022   
17                     Parque de los Niños     CA041    23/2/2022   
27                           Náutico 400 m     QU062    23/2/2022   
30                Puerto Trinidad calle 47     BZ077    23/2/2022   
52                       Espigón La Farola     SI021     4/5/2022   
65                          Arroyo Sarandí     AV052     4/5/2022   
66                 Costa de Villa Domínico     AV055     4/5/2022   
67                    Arroyo Santo Domingo     AV053     4/5/2022   
94                       Espigón La Farola     SI021    23/8/2022   
95                       Reserva Ecológica     SI022    23/8/2022   
96                Playa Espigón de Pacheco     SI024    23/8/2022   
97                             Perú Puente     SI023    23/8/2022   
101                    Parque de los Niños     CA041    23/8/2022   
123                     Playa La Bagliardi     BS091    23/8/2022   
126           Canal Villanueva y Río Luján     TI001   31/10/2022   
127           Río Lujan y Arroyo Caraguatá     TI006   31/10/2022   
128            Canal Aliviador y Río Lujan     TI002   31/10/2022   
129  Río Carapachay y Arroyo Gallo Fiambre     TI003   31/10/2022   
130            Río Reconquista y Río Lujan     TI004   31/10/2022   
131     Rio Tigre 100m antes del Rio Luján     TI005   31/10/2022   
132         Río Lujan y Canal San Fernando     TI007   31/10/2022   
133          Río Capitán y Río San Antonio     TI008   31/10/2022   
134         Arroyo Abra Vieja y Santa Rosa     TI009   31/10/2022   
135                               Del Arca     SF015   31/10/2022   
136                      Espigón La Farola     SI021  no se midió   
137                      Reserva Ecológica     SI022   31/10/2022   
138               Playa Espigón de Pacheco     SI024   31/10/2022   
139                            Perú Puente     SI023   31/10/2022   
140                Reserva Barrio El Ceibo     VL033   31/10/2022   
141               Puerto de Olivos Espigón     VL032   31/10/2022   
142                           Costa y Melo     VL031   31/10/2022   
147                Escollera de Propaneros     AV054   31/10/2022   
148             Polo Petroquímico Dock Sud     AV051   31/10/2022   
149                         Arroyo Sarandí     AV052   31/10/2022   
150                Costa de Villa Domínico     AV055   31/10/2022   
151                   Arroyo Santo Domingo     AV053   31/10/2022   
152                           Espora 400 m     QU061   31/10/2022   
153                          Náutico 400 m     QU062   31/10/2022   
154                    Pejerrey Club 400 m     QU063   31/10/2022   
155       Calle 14 y Costa - Salida cloaca     BZ078   31/10/2022   
156               Puerto Trinidad calle 47     BZ077   31/10/2022   
157              Costanera Hudson Calle 63     BZ080   31/10/2022   
158     Boca Cerrada (Res.Nat. Punta Lara)  EN-extra   31/10/2022   
159                      Camping Eva Perón     EN081   31/10/2022   
160             Toma de agua Club de Pesca     EN082   31/10/2022   
161                         Arroyo El Gato     EN083   31/10/0202   
162      Ensenada Prefectura Isla Santiago     EN084   31/10/0202   
163                  Balneario Palo Blanco     BS092   31/10/2022   
164          Diagonal 66 (descarga cloaca)     BS095   31/10/2022   
165                     Playa La Bagliardi     BS091   31/10/2022   
166                    Balneario Municipal     BS094   31/10/2022   
167                      Playa La Balandra     BS093   31/10/2022   

         campaña     tem_agua     tem_aire           od           ph  \
10        Verano  no se midió  no se midió  no se midió  no se midió   
12        Verano         26.5           27          5.3         7.49   
17        Verano         26.1  no se midió          4.8          7.4   
27        Verano     

Como pudimos ver hay muchisimos "no se midió" que representa NA, entonces vamos a convertir para todas las columnas
el "no se midió" a NA

In [15]:
lista_columnas = copy_dataset.columns

for col in lista_columnas:
    copy_dataset[col] = copy_dataset[col].replace(['no midieron este día', 'no se midió', 'no se determinó'], pd.NA)
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire    od    ph     olores      color  ... dbo_mg_l  \
0       24.5     23.3   5.3  6.56   Ausencia   Ausencia  ...      6.2   
1       25.4     23.3  2.25  6.56  Presencia  Presencia  ...      5.8   
2       24.6     23.3  2.94  6.59   Ausencia  Presencia  ...      1.9   
3       25.2     23.3  2.22  7.45  Presencia  Presencia  ...      5.8   
4       24.1       20  1.02  6.39   Ausencia  Presencia  ...      2.6   
..       ...      ...   ...   ...        ...        ...  ...      ...   
163       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
164       10       12  <NA>  <NA>   Ausencia  Presencia  ...     <NA>   
165       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
166       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
167       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                <0.10        <0.005        <0.001   
1         29            34                <0.10        <0.005        <0.001   
2         29            17                <0.10        <0.005        <0.001   
3         29            23                <0.10        <0.005        <0.001   
4         59            18                <0.10        <0.005        <0.001   
..       ...           ...                  ...           ...           ...   
163      <30            23                <0.10        <0.005        <0.001   
164       31          18.2                <0.10        <0.005        <0.001   
165      <30            40                <0.10        <0.005        <0.001   
166       39            90                <0.10             5        <0.001   
167      <30            70                <0.10             5        <0.001   

    clorofila_a_ug_l microcistina_ug_l ica             calidad_de_agua  
0                <10             <0.20  55             Muy deteriorada  
1                <10             <0.20  42  Extremadamente deteriorada  
2                <10               0.2  45             Muy deteriorada  
3                <10             <0.20  46             Muy deteriorada  
4                <10             <0.20  44  Extremadamente deteriorada  
..               ...               ...  ..                         ...  
163              2.1             <0.15  43  Extremadamente deteriorada  
164             20.2             <0.15  37  Extremadamente deteriorada  
165              0.2             <0.15  49             Muy deteriorada  
166             10.5             <0.15  39  Extremadamente deteriorada  
167             48.0             <0.15  34  Extremadamente deteriorada  

[168 rows x 29 columns]

Una vez pasado todos los valores a NA vamos a contar cuantos NA tenemos por cada variable

In [16]:
copy_dataset.isna().sum()

sitios                      0
codigo                      0
fecha                       5
campaña                     5
tem_agua                   23
tem_aire                   24
od                         28
ph                         28
olores                     13
color                      13
espumas                    13
mat_susp                   13
colif_fecales_ufc_100ml    15
escher_coli_ufc_100ml      15
enteroc_ufc_100ml          15
nitrato_mg_l               14
nh4_mg_l                   14
p_total_l_mg_l             21
fosf_ortofos_mg_l          14
dbo_mg_l                   58
dqo_mg_l                   14
turbiedad_ntu              14
hidr_deriv_petr_ug_l       18
cr_total_mg_l              17
cd_total_mg_l              17
clorofila_a_ug_l           18
microcistina_ug_l          18
ica                        14
calidad_de_agua            14
dtype: int64

porcentaje de nulls

In [17]:
copy_dataset.isna().sum() / copy_dataset.shape[0]

sitios                     0.000000
codigo                     0.000000
fecha                      0.029762
campaña                    0.029762
tem_agua                   0.136905
tem_aire                   0.142857
od                         0.166667
ph                         0.166667
olores                     0.077381
color                      0.077381
espumas                    0.077381
mat_susp                   0.077381
colif_fecales_ufc_100ml    0.089286
escher_coli_ufc_100ml      0.089286
enteroc_ufc_100ml          0.089286
nitrato_mg_l               0.083333
nh4_mg_l                   0.083333
p_total_l_mg_l             0.125000
fosf_ortofos_mg_l          0.083333
dbo_mg_l                   0.345238
dqo_mg_l                   0.083333
turbiedad_ntu              0.083333
hidr_deriv_petr_ug_l       0.107143
cr_total_mg_l              0.101190
cd_total_mg_l              0.101190
clorofila_a_ug_l           0.107143
microcistina_ug_l          0.107143
ica                        0

In [22]:
columnas = [
    "nh4_mg_l",
    "p_total_l_mg_l",
    "fosf_ortofos_mg_l",
    "dbo_mg_l",
    "dqo_mg_l",
    "turbiedad_ntu",
    "hidr_deriv_petr_ug_l",
    "cr_total_mg_l",
    "cd_total_mg_l",
    "clorofila_a_ug_l",
    "microcistina_ug_l"
]
for col in columnas:
    copy_dataset[col] = copy_dataset[col].str.strip("<")
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire    od    ph     olores      color  ... dbo_mg_l  \
0       24.5     23.3   5.3  6.56   Ausencia   Ausencia  ...      6.2   
1       25.4     23.3  2.25  6.56  Presencia  Presencia  ...      5.8   
2       24.6     23.3  2.94  6.59   Ausencia  Presencia  ...      1.9   
3       25.2     23.3  2.22  7.45  Presencia  Presencia  ...      5.8   
4       24.1       20  1.02  6.39   Ausencia  Presencia  ...      2.6   
..       ...      ...   ...   ...        ...        ...  ...      ...   
163       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
164       10       12  <NA>  <NA>   Ausencia  Presencia  ...     <NA>   
165       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
166       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
167       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                 0.10         0.005         0.001   
1         29            34                 0.10         0.005         0.001   
2         29            17                 0.10         0.005         0.001   
3         29            23                 0.10         0.005         0.001   
4         59            18                 0.10         0.005         0.001   
..       ...           ...                  ...           ...           ...   
163       30            23                 0.10         0.005         0.001   
164       31          18.2                 0.10         0.005         0.001   
165       30            40                 0.10         0.005         0.001   
166       39            90                 0.10             5         0.001   
167       30            70                 0.10             5         0.001   

    clorofila_a_ug_l microcistina_ug_l ica             calidad_de_agua  
0                 10              0.20  55             Muy deteriorada  
1                 10              0.20  42  Extremadamente deteriorada  
2                 10               0.2  45             Muy deteriorada  
3                 10              0.20  46             Muy deteriorada  
4                 10              0.20  44  Extremadamente deteriorada  
..               ...               ...  ..                         ...  
163              2.1              0.15  43  Extremadamente deteriorada  
164             20.2              0.15  37  Extremadamente deteriorada  
165              0.2              0.15  49             Muy deteriorada  
166             10.5              0.15  39  Extremadamente deteriorada  
167             48.0              0.15  34  Extremadamente deteriorada  

[168 rows x 29 columns]

In [23]:
niveles_deterioro = {
    "Deteriorada" : 1,
    "Muy deteriorada" : 2,
    "Extremadamente deteriorada" : 3
}

copy_dataset["calidad_de_agua"] = copy_dataset["calidad_de_agua"].map(niveles_deterioro)
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire    od    ph     olores      color  ... dbo_mg_l  \
0       24.5     23.3   5.3  6.56   Ausencia   Ausencia  ...      6.2   
1       25.4     23.3  2.25  6.56  Presencia  Presencia  ...      5.8   
2       24.6     23.3  2.94  6.59   Ausencia  Presencia  ...      1.9   
3       25.2     23.3  2.22  7.45  Presencia  Presencia  ...      5.8   
4       24.1       20  1.02  6.39   Ausencia  Presencia  ...      2.6   
..       ...      ...   ...   ...        ...        ...  ...      ...   
163       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
164       10       12  <NA>  <NA>   Ausencia  Presencia  ...     <NA>   
165       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
166       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   
167       10       12  <NA>  <NA>   Ausencia   Ausencia  ...     <NA>   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                 0.10         0.005         0.001   
1         29            34                 0.10         0.005         0.001   
2         29            17                 0.10         0.005         0.001   
3         29            23                 0.10         0.005         0.001   
4         59            18                 0.10         0.005         0.001   
..       ...           ...                  ...           ...           ...   
163       30            23                 0.10         0.005         0.001   
164       31          18.2                 0.10         0.005         0.001   
165       30            40                 0.10         0.005         0.001   
166       39            90                 0.10             5         0.001   
167       30            70                 0.10             5         0.001   

    clorofila_a_ug_l microcistina_ug_l ica calidad_de_agua  
0                 10              0.20  55             2.0  
1                 10              0.20  42             3.0  
2                 10               0.2  45             2.0  
3                 10              0.20  46             2.0  
4                 10              0.20  44             3.0  
..               ...               ...  ..             ...  
163              2.1              0.15  43             3.0  
164             20.2              0.15  37             3.0  
165              0.2              0.15  49             2.0  
166             10.5              0.15  39             3.0  
167             48.0              0.15  34             3.0  

[168 rows x 29 columns]

In [29]:
copy_dataset["mat_susp"].value_counts()

mat_susp
Ausencia     106
Presencia     26
Ausente       16
ausencia       6
presencia      1
Name: count, dtype: int64

In [32]:
mapeo_dicotomico = {
    "Ausencia" : 0,
    "Ausente" : 0,
    "ausencia" : 0,
    "presencia" : 1,
    "Presencia" : 1
}

copy_dataset["olores"] = copy_dataset["olores"].map(mapeo_dicotomico)
copy_dataset["color"] = copy_dataset["color"].map(mapeo_dicotomico)
copy_dataset["espumas"] = copy_dataset["espumas"].map(mapeo_dicotomico)
copy_dataset["mat_susp"] = copy_dataset["mat_susp"].map(mapeo_dicotomico)
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire    od    ph  olores  color  ...  dbo_mg_l  dqo_mg_l  \
0       24.5     23.3   5.3  6.56     0.0    0.0  ...       6.2        29   
1       25.4     23.3  2.25  6.56     1.0    1.0  ...       5.8        29   
2       24.6     23.3  2.94  6.59     0.0    1.0  ...       1.9        29   
3       25.2     23.3  2.22  7.45     1.0    1.0  ...       5.8        29   
4       24.1       20  1.02  6.39     0.0    1.0  ...       2.6        59   
..       ...      ...   ...   ...     ...    ...  ...       ...       ...   
163       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   
164       10       12  <NA>  <NA>     0.0    1.0  ...      <NA>        31   
165       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   
166       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        39   
167       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   

    turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0              90                 0.10         0.005         0.001   
1              34                 0.10         0.005         0.001   
2              17                 0.10         0.005         0.001   
3              23                 0.10         0.005         0.001   
4              18                 0.10         0.005         0.001   
..            ...                  ...           ...           ...   
163            23                 0.10         0.005         0.001   
164          18.2                 0.10         0.005         0.001   
165            40                 0.10         0.005         0.001   
166            90                 0.10             5         0.001   
167            70                 0.10             5         0.001   

    clorofila_a_ug_l microcistina_ug_l ica calidad_de_agua  
0                 10              0.20  55             2.0  
1                 10              0.20  42             3.0  
2                 10               0.2  45             2.0  
3                 10              0.20  46             2.0  
4                 10              0.20  44             3.0  
..               ...               ...  ..             ...  
163              2.1              0.15  43             3.0  
164             20.2              0.15  37             3.0  
165              0.2              0.15  49             2.0  
166             10.5              0.15  39             3.0  
167             48.0              0.15  34             3.0  

[168 rows x 29 columns]

In [33]:
copy_dataset["fecha"].value_counts()

fecha
23/2/2022     42
4/5/2022      42
23/8/2022     42
31/10/2022    35
31/10/0202     2
Name: count, dtype: int64

In [34]:
copy_dataset["fecha"] = copy_dataset["fecha"].replace("31/10/0202", "31/10/2022")
copy_dataset["fecha"].value_counts()

fecha
23/2/2022     42
4/5/2022      42
23/8/2022     42
31/10/2022    37
Name: count, dtype: int64

In [35]:
copy_dataset

sitios codigo       fecha    campaña  \
0             Canal Villanueva y Río Luján  TI001   23/2/2022     Verano   
1             Río Lujan y Arroyo Caraguatá  TI006   23/2/2022     Verano   
2              Canal Aliviador y Río Lujan  TI002   23/2/2022     Verano   
3    Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022     Verano   
4              Río Reconquista y Río Lujan  TI004   23/2/2022     Verano   
..                                     ...    ...         ...        ...   
163                  Balneario Palo Blanco  BS092  31/10/2022  Primavera   
164          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  Primavera   
165                     Playa La Bagliardi  BS091  31/10/2022  Primavera   
166                    Balneario Municipal  BS094  31/10/2022  Primavera   
167                      Playa La Balandra  BS093  31/10/2022  Primavera   

    tem_agua tem_aire    od    ph  olores  color  ...  dbo_mg_l  dqo_mg_l  \
0       24.5     23.3   5.3  6.56     0.0    0.0  ...       6.2        29   
1       25.4     23.3  2.25  6.56     1.0    1.0  ...       5.8        29   
2       24.6     23.3  2.94  6.59     0.0    1.0  ...       1.9        29   
3       25.2     23.3  2.22  7.45     1.0    1.0  ...       5.8        29   
4       24.1       20  1.02  6.39     0.0    1.0  ...       2.6        59   
..       ...      ...   ...   ...     ...    ...  ...       ...       ...   
163       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   
164       10       12  <NA>  <NA>     0.0    1.0  ...      <NA>        31   
165       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   
166       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        39   
167       10       12  <NA>  <NA>     0.0    0.0  ...      <NA>        30   

    turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0              90                 0.10         0.005         0.001   
1              34                 0.10         0.005         0.001   
2              17                 0.10         0.005         0.001   
3              23                 0.10         0.005         0.001   
4              18                 0.10         0.005         0.001   
..            ...                  ...           ...           ...   
163            23                 0.10         0.005         0.001   
164          18.2                 0.10         0.005         0.001   
165            40                 0.10         0.005         0.001   
166            90                 0.10             5         0.001   
167            70                 0.10             5         0.001   

    clorofila_a_ug_l microcistina_ug_l ica calidad_de_agua  
0                 10              0.20  55             2.0  
1                 10              0.20  42             3.0  
2                 10               0.2  45             2.0  
3                 10              0.20  46             2.0  
4                 10              0.20  44             3.0  
..               ...               ...  ..             ...  
163              2.1              0.15  43             3.0  
164             20.2              0.15  37             3.0  
165              0.2              0.15  49             2.0  
166             10.5              0.15  39             3.0  
167             48.0              0.15  34             3.0  

[168 rows x 29 columns]